test/junit/scala/collection/StringParsersTest.scala

```diff
@@ -0,0 +1,332 @@
+    "0x0.00000000000019p-1022",
+    "0x0.000000000000190000000000000000001p-1022",
+    "0x0.00000000000019fffffffffffffffffffffffffffffffffp-1022",
+    "0x0.0000000000001fp-1022",
+    "0x0.0000000000001f0000000000000000001p-1022",
+    "0x0.0000000000001ffffffffffffffffffffffffffffffffffp-1022",
+    "-0x0.0000000000001p-1022",
+    "-0x0.00000000000010000000000000000001p-1022",
+    "-0x0.0000000000001fffffffffffffffffffffffffffffffffp-1022",
+    "-0x0.00000000000017p-1022",
+    "-0x0.000000000000170000000000000000001p-1022",
+    "-0x0.00000000000017fffffffffffffffffffffffffffffffffp-1022",
+    "-0x0.00000000000018p-1022",
+    "-0x0.000000000000180000000000000000001p-1022",
+    "-0x0.00000000000018fffffffffffffffffffffffffffffffffp-1022",
+    "-0x0.00000000000019p-1022",
+    "-0x0.000000000000190000000000000000001p-1022",
+    "-0x0.00000000000019fffffffffffffffffffffffffffffffffp-1022",
+    "-0x0.0000000000001fp-1022",
+    "-0x0.0000000000001f0000000000000000001p-1022",
+    "-0x0.0000000000001ffffffffffffffffffffffffffffffffffp-1022",
+    "0x0.00000000000004p-1022",
+    "0x0.00000000000007ffffffffffffffffffffffp-1022",
+    "0x0.00000000000008p-1022",
+    "0x0.000000000000080000000000000000001p-1022",
+    "0x0.00000000000008fffffffffffffffffffffffffffffffp-1022",
+    "0x0.00000000000009p-1022",
+    "0x0.000000000000090000000000000000001p-1022",
+    "0x0.00000000000009fffffffffffffffffffffffffffffffffp-1022",
+    "0x0.0000000000000fffffffffffffffffffffffffffffffffffp-1022",
+    "-0x0.00000000000004p-1022",
+    "-0x0.00000000000007ffffffffffffffffffffffp-1022",
+    "-0x0.00000000000008p-1022",
+    "-0x0.000000000000080000000000000000001p-1022",
+    "-0x0.00000000000008fffffffffffffffffffffffffffffffp-1022",
+    "-0x0.00000000000009p-1022",
+    "-0x0.000000000000090000000000000000001p-1022",
+    "-0x0.00000000000009fffffffffffffffffffffffffffffffffp-1022",
+    "-0x0.0000000000000fffffffffffffffffffffffffffffffffffp-1022",
+    "0.0",
+    "+0.0",
+    "-0.0",
+    "NaN",
+    "+NaN",
+    "-NaN",
+    "Infinity",
+    "+Infinity",
+    "-Infinity",
+    "NaNd",
+    "+NaNd",
+    "-NaNd",
+    "Infinityd",
+    "+Infinityd",
+    "-Infinityd"
+  )
+
+  @Test
+  def doubleSpecificTest: Unit = doubleExamples.foreach(doubleOK)
+
+  @Test
+  def doubleGeneralTest: Unit = forAllExamples.foreach(doubleOK)
+
+  @Test
+  def floatSpecificTest: Unit = doubleExamples.foreach(floatOK)
+
+  @Test
+  def floatGeneralTest: Unit = forAllExamples.foreach(floatOK)
+
+  @Test
+  def byteTest: Unit = (forAllExamples ::: nearOverflow).foreach(byteOK)
+
+  @Test
+  def shortTest: Unit = (forAllExamples ::: nearOverflow).foreach(shortOK)
+
+  @Test
+  def intTest: Unit = (forAllExamples ::: nearOverflow).foreach(intOK)
+
+  @Test
+  def longTest: Unit = (forAllExamples ::: longNearOverflow).foreach(longOK)
+
+  @Test
+  def nullByte: Unit = try {
+    nullstring.parseByte
+    Assert.fail("null.parseByte throws NPE")
```

Comment:
You could use our `assertThrows` (https://github.com/scala/scala/blob/v2.13.0-M4/test/junit/scala/tools/testing/AssertUtil.scala#L44)